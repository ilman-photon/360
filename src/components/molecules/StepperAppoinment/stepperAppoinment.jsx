import * as React from "react";
import Box from "@mui/material/Box";
import Stepper from "@mui/material/Stepper";
import Step from "@mui/material/Step";
import StepLabel from "@mui/material/StepLabel";
import { useMediaQuery } from "@mui/material";

export default function StepperAppoinment({ ...props }) {
  const steps = props.steps;
  const isDesktop = useMediaQuery("(min-width: 769px)");

  return (
    <Box
      sx={{
        width: "100%",
        display: "flex",
        p: { xs: 1, md: "16px 240px" },
        mt: { xs: 0, md: "80px" },
        justifyContent: "center",
        borderBottom: "1px solid #E0E0E0",
      }}
    >
      <Stepper
        activeStep={props.activeStep}
        alternativeLabel
        sx={{ width: "800px" }}
      >
        {steps.map((label, idx) => (
          <Step sx={{ color: "#87C9D2" }} key={label}>
            <StepLabel
              sx={{
                ".css-qivjh0-MuiStepLabel-label.MuiStepLabel-alternativeLabel":
                  {
                    marginTop: 1,
                    color: "#000000",
                  },
              }}
            >
              {isDesktop ? label : idx === props.activeStep ? label : null}
            </StepLabel>
          </Step>
        ))}
      </Stepper>
    </Box>
  );
}
