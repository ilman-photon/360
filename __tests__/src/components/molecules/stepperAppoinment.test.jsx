import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import StepperAppoinment from "../../../../src/components/molecules/StepperAppoinment/stepperAppoinment";
import mediaQuery from 'css-mediaquery';

describe("StepperAppoinment Components", () => {
  let container;
  const mockSteps = [
    "Location",
    "Review",
    "Appointment Details",
    "Contact Info",
    "Confirm",
  ];

  function createMatchMedia(width) {
    return query => ({
        matches: mediaQuery.match(query, { width }),
        addListener: () => { },
        removeListener: () => { },
    });
  }

  beforeEach(() => {
    window.matchMedia = createMatchMedia('1920px');
    container = render(<StepperAppoinment activeStep={1} steps={mockSteps} />);
  });

  it("StepperAppoinment render (mobile)", () => {
    expect(container).toMatchSnapshot();
    expect(container.getByText("Location")).toBeInTheDocument();
    expect(container.getByText("Review")).toBeInTheDocument();
    expect(container.getByText("Appointment Details")).toBeInTheDocument();
    expect(container.getByText("Contact Info")).toBeInTheDocument();
    expect(container.getByText("Confirm")).toBeInTheDocument();
    expect(container.getByText("3")).toBeInTheDocument();
    expect(container.getByText("4")).toBeInTheDocument();
    expect(container.getByText("5")).toBeInTheDocument();
  });
});
