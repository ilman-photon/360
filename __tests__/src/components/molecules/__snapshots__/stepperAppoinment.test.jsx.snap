// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepperAppoinment Components StepperAppoinment render (mobile) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-17a7zap"
      >
        <div
          class="MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel css-1fqle5r-MuiStepper-root"
        >
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel Mui-completed css-pf7qci-MuiStep-root"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-completed css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-pf7qci-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-active css-zpcwqm-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-active MuiStepLabel-alternativeLabel css-qivjh0-MuiStepLabel-label"
                >
                  Review
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-pf7qci-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-pf7qci-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    4
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-pf7qci-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    5
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-17a7zap"
    >
      <div
        class="MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel css-1fqle5r-MuiStepper-root"
      >
        <div
          class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel Mui-completed css-pf7qci-MuiStep-root"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
          >
            <span
              class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-completed css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            />
          </span>
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-pf7qci-MuiStep-root"
        >
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-active css-zpcwqm-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
            />
          </div>
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
          >
            <span
              class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  2
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="MuiStepLabel-label Mui-active MuiStepLabel-alternativeLabel css-qivjh0-MuiStepLabel-label"
              >
                Review
              </span>
            </span>
          </span>
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-pf7qci-MuiStep-root"
        >
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
            />
          </div>
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
          >
            <span
              class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  3
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            />
          </span>
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-pf7qci-MuiStep-root"
        >
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
            />
          </div>
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
          >
            <span
              class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  4
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            />
          </span>
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-pf7qci-MuiStep-root"
        >
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
            />
          </div>
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-1vlz8yb-MuiStepLabel-root"
          >
            <span
              class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  5
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
